extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/attribute/RemoteUserAttribute.java

```diff
@@ -20,11 +21,15 @@ private RemoteUserAttribute() {
 
     @Override
     public String readAttribute(final RoutingContext exchange) {
-        QuarkusHttpUser sc = (QuarkusHttpUser) exchange.user();
-        if (sc == null) {
+        User user = (User) exchange.user();
```

Comment:
Do you actually need a cast here?

Reply:
Thanks for noticing it @gsmet, indeed, not needed

Reply:
I moved `username` to the constant declaration along other ones there